Automattic/wp-calypso
# composite checkout - add plan length picker to review step

I recommend looking at the commits one at a time.

#### Changes proposed in this Pull Request

* Adds radio buttons to the order review step of composite checkout allowing users to change the plan term length.

To do:

- [x] Use a state machine to enable immediate feedback

#### Testing instructions

```
npm run test-client client/my-sites/checkout/checkout
```

Manual testing:
1. Enter checkout with a plan
2. Proceed to the review order step
3. Verify that radio buttons appear corresponding to the plan term lengths (1 year and 2 years), and that the currently selected term is correct.
4. Click to select a different term length; verify that the selected radio button and the prices (line item and total) update.

Edge cases:
1. Change the plan length, then change it again before the first change takes effect.
2. Switch between plan lengths several times in succession

Example:

<img width="602" alt="image" src="https://user-images.githubusercontent.com/9310939/73416280-060c6200-42da-11ea-870d-607f98701666.png">

<img width="604" alt="image" src="https://user-images.githubusercontent.com/9310939/73416286-0c024300-42da-11ea-9b1f-37570967de68.png">
